tests/test_fit_result.py

```diff
@@ -1,219 +1,333 @@
-        assert not hasattr(self.fit_result, 'log_likelihood')
-        assert not hasattr(self.fit_result, 'likelihood')
-
-        assert hasattr(self.minpack_result, 'objective_value')
-        assert hasattr(self.minpack_result, 'r_squared')
-        assert hasattr(self.minpack_result, 'chi_squared')
-        assert not hasattr(self.minpack_result, 'log_likelihood')
-        assert not hasattr(self.minpack_result, 'likelihood')
-
-        assert hasattr(self.likelihood_result, 'objective_value')
-        assert not hasattr(self.likelihood_result, 'r_squared')
-        assert not hasattr(self.likelihood_result, 'chi_squared')
-        assert hasattr(self.likelihood_result, 'log_likelihood')
-        assert hasattr(self.likelihood_result, 'likelihood')
+def _run_tests(fit_result, expected_par, expected_std, expected_obj, expected_gof):
+    """
+    This method compares a FitResult with 
+    the expected values of the parameters, standard deviation, 
+    minimizer, objectives and goodness of fit estimators.
+
+    :param fit_result: Test-instance of FitResult 
+    :param expected_par: dict of parameters with expected value
+    :param expected_std: dict of parameters with expected standard deviation
+    :param expected_obj: expected Class of objective
+    :param expected_gof: dict of fit estimators and expected value.
+    """
+
+    assert isinstance(fit_result.params, OrderedDict)
+    assert isinstance(fit_result.minimizer_output, dict)
+    assert isinstance(fit_result, FitResults)
+
+    for attr_name, value in expected_par.items():
+        assert fit_result.value(attr_name) == pytest.approx(value, abs=1e-5)
+
+    for attr_name, value in expected_std.items():
+        assert fit_result.stdev(attr_name) == pytest.approx(value, abs=1e-5, nan_ok=True)
+
+    assert isinstance(fit_result.minimizer, BaseMinimizer)
+    assert isinstance(fit_result.objective, expected_obj)
+    assert isinstance(fit_result.status_message, str)
+
+    dumped = pickle.dumps(fit_result)
+    new_result = pickle.loads(dumped)
+    assert sorted(fit_result.__dict__.keys()) == sorted(new_result.__dict__.keys())
+    for k, v1 in fit_result.__dict__.items():
+        v2 = new_result.__dict__[k]
+        if k == 'minimizer':
+            assert type(v1) == type(v2)
+        elif k != 'minimizer_output':  # Ignore minimizer_output
+            if isinstance(v1, np.ndarray):
+                assert v1 == pytest.approx(v2, nan_ok=True)
+
+    if fit_result.constraints:
+        assert isinstance(fit_result.constraints, list)
+        for constraint in fit_result.constraints:
+            assert isinstance(constraint, MinimizeModel)
+
+    for attr_name, value in expected_gof.items():
+        assert getattr(fit_result, attr_name) == pytest.approx(value,abs=1e-5, nan_ok=True)
+
+
+@pytest.mark.parametrize('minimizer',
+        # Removed TrustConstr and COBYLA because their results differ greatly #TODO add them if fixed
```

Comment:
not anymore :D thank you!

Reply:
Not anymore 🙌 The results of COBYLA and TrustConstr are correct with more iterations.

Reply:
Edit: TrustConstr runs with LogLikelihood, 1e8 as the maximum number of iterations and no constraints forever. I think that's because of the LL, but we should keep an eye on it.